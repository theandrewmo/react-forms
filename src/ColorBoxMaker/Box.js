/** Box - displays a div with a background color, width and height based on props
 * 
 * Props: {key, backgroundColor, width, height, removeBox}
 * 
 * State: none
 * 
 */

const Box = ({key, backgroundColor, width, height, removeBox}) => {
    const boxStyle = {
        backgroundColor,
        width,
        height 
       }
    return (
        <div style={{ display: 'flex' }} >  
            <div id={key} style={boxStyle} data-testid={`box-${backgroundColor}-${width}-${height}`}></div>
            <button onClick={removeBox} style={{height: '30px', width: '30px'}}>X</button>
        </div>
    )
}

export default Box;