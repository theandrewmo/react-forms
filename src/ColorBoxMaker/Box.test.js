import { render } from '@testing-library/react';
import Box from './Box';

it('renders', () => {
  render(<Box />);
});

it("matches snapshot", () => {
    const { asFragment } = render(<Box 
        backgroundColor='red'
        width='100px'
        height='100px'
        />);
    expect(asFragment()).toMatchSnapshot();
})



