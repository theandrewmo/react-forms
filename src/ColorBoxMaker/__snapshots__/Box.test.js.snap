// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    style="display: flex;"
  >
    <div
      data-testid="box-red-100px-100px"
      style="background-color: red; width: 100px; height: 100px;"
    />
    <button
      style="height: 30px; width: 30px;"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
